build.gradle

```diff
@@ -160,6 +160,52 @@ subprojects {
         maven { url "https://ci.opensearch.org/ci/dbc/snapshots/lucene/" }
         maven { url 'https://jitpack.io' }
     }
+
+    // Publish internal modules as Maven artifacts for external use, such as by opensearch-spark and opensearch-cli.
+    def publishedModules = ['sql', 'ppl', 'core', 'opensearch', 'common', 'protocol']
+    if (publishedModules.contains(name)) {
+        apply plugin: 'java-library'
+        apply plugin: 'maven-publish'
+
+        def fullArtifactId = "unified-query-${project.name}"
+        publishing {
+            publications {
+                unifiedQuery(MavenPublication) {
+                    from components.java
+                    groupId = "org.opensearch.sql"
```

Comment:
could we use "org.opensearch.ql", "org.opensearch.query"?

Reply:
Sure, addressed in https://github.com/opensearch-project/sql/pull/3763/commits/b14a5d11ccc9ff28e71c92e2ac61f2f6c35ee2b9.